attn_gradcheck.py

```diff
@@ -0,0 +1,45 @@
+import torch
+
+class Attn(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, q, k, v):
+        attn = torch.matmul(q, k.swapaxes(0, 1))
+        attn = torch.tanh(attn)
+        output = torch.matmul(attn, v)
+
+        ctx.save_for_backward(q, k, v, attn)
+        return output, attn
+
+    @staticmethod
+    def backward(ctx, grad_output, grad_attn):
```

Comment:
Note that you could be using `needs_input_grad` here to make the backward faster in the case where not all inputs to the forward have requires_grad = True